src/sqlfluff/rules/L043.py

```diff
@@ -90,20 +92,19 @@ def _eval(self, segment, **kwargs):
                 # If then-false, add "not" and space
                 edits = []
                 if then_bool_type == "FALSE":
-                    not_space = [
-                        self.make_keyword(raw="not"),
-                        self.make_whitespace(raw=" "),
-                    ]
-                    edits.extend(not_space)
+                    edits.extend(
+                        [
+                            KeywordSegment("not"),
+                            WhitespaceSegment(),
+                        ]
+                    )
                 # Add coalesce and parenthesis
-                coalesce_parenthesis = [
-                    self.make_keyword(raw="coalesce"),
-                    self.make_symbol(
-                        raw="(",
-                        seg_type="parenthesis",
-                    ),
-                ]
-                edits.extend(coalesce_parenthesis)
+                edits.extend(
+                    [
+                        KeywordSegment("coalesce"),
+                        SymbolSegment("(", name="start_bracket", type="start_bracket"),
```

Comment:
ah - I think the `parenthesis` segments before wouldn't actually work in bracket matching. I don't think it had caused a bug yet, but I think it might in future.